src/rantanplan/structures.py

```diff
@@ -231,6 +231,32 @@ def is_ovillejo(lengths):
     return correct_lines == number_of_verses
 
 
+def rank_character_occurrences(string):
+    """
+    Count how many times does a character occur in a given string before its
+    own position.
+    :param string: String with the rhyme pattern
+    :return: A string with the count values
+    """
+    count_dict = {}
+    output = []
+    for character in string:
+        count = count_dict.get(character, 0)
+        output.append(str(count))
```

Comment:
Any reason why this function needs to return the counts as strings? I know it's used in the main function, but maybe it'd be better to return numbers and have the main function do what it needs with the counts?